packages/upset/src/components/AltTextSidebar.tsx

```diff
@@ -75,65 +65,42 @@ export const AltTextSidebar: FC<Props> = ({ open, close, generateAltText }) => {
         setTextGenErr(false);
       } catch (e) {
         const msg: string = (e as Error).message;
-        // We want the error message to display on the frontend
-        setAltText({
-          longDescription: msg,
-          shortDescription: msg,
-          techniqueDescription: msg,
-        });
-        setTextGenErr(true);
+        setTextGenErr(msg);
       }
     }
-
+    
     generate();
   }, [currState]);
 
-  // this useEffect resets the plot information when the edit is toggled off
-  useEffect(() => {
-    if (!isEditable) {
-      setPlotInformation(plotInformationState);
+  // Current alt text to display to the user
+  const displayAltText: string | undefined = useMemo(() => {
+    if (textEditing) {
+      return useLong
+      ? userLongText ?? altText?.longDescription
+      : userShortText ?? altText?.shortDescription
```

Comment:
The ternary mixed with the `??` operator is confusing. Either making this more clear or documenting the logic here may be helpful for readability.